idaes/gas_solid_contactors/unit_models/fixed_bed_0D.py

```diff
@@ -0,0 +1,330 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+IDAES 0D Fixed Bed Reactor model.
+"""
+
+# Import Pyomo libraries
+from pyomo.environ import Constraint, Var
+from pyomo.dae import DerivativeVar
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import (declare_process_block_class,
+                        EnergyBalanceType,
+                        UnitModelBlockData,
+                        useDefault)
+from idaes.core.util.config import (is_physical_parameter_block)
+import idaes.logger as idaeslog
+from idaes.core.util import get_solver
+
+__author__ = "Chinedu Okoli, Andrew Lee"
+
+
+# Assumptions:
+# Only Vap and Sol phases, which are explicitly named as such
+# Perfect mixing in Vap phase
+# Static solid phase
+
+# Need to have build-on-demand properties in reactions for this to work
+
+@declare_process_block_class("FixedBed0D")
+class FixedBed0DData(UnitModelBlockData):
+    """
+    0D Fixed Bed Reactor Model Class
+    """
+    CONFIG = ConfigBlock()
+    CONFIG.declare("dynamic", ConfigValue(
+        domain=In([True]),
+        default=True,
+        description="Dynamic model flag - must be True",
+        doc="""Indicates whether this model will be dynamic or not,
+**default** = True. Fixed beds must be dynamic."""))
+    CONFIG.declare("has_holdup", ConfigValue(
+        default=True,
+        domain=In([True]),
+        description="Holdup construction flag - must be True",
+        doc="""Indicates whether holdup terms should be constructed or not.
+**default** - True. Fixed bed reactors must be dynamic, thus this must be
+True."""))
+    CONFIG.declare("energy_balance_type", ConfigValue(
+        default=EnergyBalanceType.useDefault,
+        domain=In(EnergyBalanceType),
+        description="Energy balance construction flag",
+        doc="""Indicates what type of energy balance should be constructed,
+**default** - EnergyBalanceType.useDefault.
```

Comment:
Update default `EnergyBalanceType.useDefault` to `EnergyBalanceType.enthalpyTotal`.